scripts/examples/gn_p6_example.sh

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+
+#
+#    Copyright (c) 2021 Project CHIP Authors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+set -e
+# Build script for GN P6 examples GitHub workflow.
+source "$(dirname "$0")/../../scripts/activate.sh"
+
+# Install required software
+if [ -d "/opt/ModusToolbox" ]; then
+    export CY_TOOLS_PATHS="/opt/ModusToolbox/tools_2.3"
+else
+    # Install tools
+    sudo apt install -y curl git make file libglib2.0-0 libusb-1.0-0 libncurses5
```

Comment:
Can we move this to docker? We should have avoid these scripts running "sudo" because developers sometimes run them manually and it is not a good idea to be installing software globally on their system in the middle of the build script and without consent (but I don't think you should add a prompt, I think it should just be moved altogether)

Reply:
Docker file already has this tools install added.. I will remove this along with Modustoolbox curl command from this script (we are expecting users to run docker before running this script ?)
Also path should be like below for compiling
export CY_TOOLS_PATHS="$HOME/sdk/modus/ModusToolbox/**tools_2.3**"

Reply:
done.. removed from gn_p6_example.sh and updated script to use different location for Modustoolbox checkout.

Could we pregenerate the JSON? Can you tell me what types of commands this needs to extract the information necessary for graph generation?
[Praveen] No.. "make getlibs" will download the assets pointed by .mtb files(which changes when there is new release) and create the JSON file dynamically based on the list of files checked out. This helps a lot in avoiding maintenance issues in the future.

Reply:
I will not approve a change that can download files during graph generation. That needs to be done ahead of time, e.g. in the bootstrap process.

Reply:
@mspang  : Added assets as git submodule as per our discussion. please review and let us know your comments.